crates/flox/src/commands/environment.rs

```diff
@@ -129,6 +132,43 @@ impl EnvironmentCommands {
                     .await?
             },
 
+            EnvironmentCommands::WipeHistory {
+                // TODO use environment_args.system?
+                environment_args: _,
+                environment,
+            } => {
+                let environment_name = environment.as_ref().map(|e| e.to_str().unwrap());
+                let environment_ref: environment_ref::EnvironmentRef =
+                    resolve_environment_ref::<GitCommandProvider>(
+                        &flox,
+                        "wipe-history",
+                        environment_name,
+                    )
+                    .await?;
+
+                let environment = environment_ref
+                    .to_env::<GitCommandProvider, NixCommandLine>(&flox)
+                    .await
+                    .context("Environment not found")?;
+
+                match environment {
+                    CommonEnvironment::Named(env) => {
+                        env.delete_symlinks().await?;
+
+                        let nix = flox.nix::<NixCommandLine>(Default::default());
+                        let store_gc = StoreGc {
+                            ..StoreGc::default()
+                        };
+
+                        info!("Running garbage collection. This may take a while...");
+                        store_gc.run(&nix, &Default::default()).await?;
+                    },
+                    CommonEnvironment::Project(_) => {
+                        bail!("can't wipe-history for project environment; project environments only keep the most recent build.");
+                    },
+                }
+            },
```

Comment:
I didn't add any integration tests, as it seems like testing garbage collection could be kinda difficult. Are unit tests good enough for this PR or should I add integration?

Reply:
I don't think that tests a whole lot more than what I'm already unit testing in https://github.com/flox/flox/pull/119/files#diff-25939f4bd2584c27f6ccd5ada240b33e03c5b6290beda0b424f37cb60f1fda02R503

I guess it would additionally check that metadata.json contains what I expect it to and we're correctly using `owner_dir()` in both bash and Rust. And I'd somehow have to disable the `nix store gc` command?

I think `nix` may set `NIX_STORE_DIR` - I briefly tried that, but `flox install` fails with `NIX_STORE_DIR` set so I don't really want to add support for that everywhere now.

Did you see the unit test and still think a more thorough test is worth it, or had you not seen the unit test?

Reply:
Created https://github.com/flox/flox/issues/125

I included a script that I'm using to test manually for reference